### PR TITLE
Fix crash when removing SpinBox during text submit

### LINKED ISSUE #77755: Crash when removing SpinBox node during text submit signal
### Godot version

4.1.dev 2e273f0e35800f2bcc7c05db54100caa97928f69

### System information

macOS 13.4

### Issue description

Removing a SpinBox from the node tree using `remove_child` during the line edit text submit signal will crash the engine.

```
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.dev.custom_build (2e273f0e35800f2bcc7c05db54100caa97928f69)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] 1   libsystem_platform.dylib            0x0000000183b5aa24 _sigtramp + 56
[2] Viewport::gui_get_focus_owner() (in godot.macos.editor.arm64) + 32
[3] SpinBox::_line_edit_focus_exit() (in godot.macos.editor.arm64) + 28
[4] CallQueue::_call_function(Callable const&, Variant const*, int, bool) (in godot.macos.editor.arm64) + 336
[5] CallQueue::flush() (in godot.macos.editor.arm64) + 908
[6] SceneTree::physics_process(double) (in godot.macos.editor.arm64) + 244
[7] Main::iteration() (in godot.macos.editor.arm64) + 528
```

### Steps to reproduce

Open the minimal reproduction project, click on the SpinBox, press enter, and the engine crashes.

### Minimal reproduction project

[SpinBoxTextSubmitCrash.zip](https://github.com/godotengine/godot/files/11631081/SpinBoxTextSubmitCrash.zip)

### PR DESCRIPTION
**Pull request by aaronfranke**
Fixes #77755.

This work was sponsored by The Mirror.

### CODE REVIEW

File: scene/gui/spin_box.cpp [hunk 1]
```diff
@@ -202,7 +202,8 @@ void SpinBox::_line_edit_focus_enter() {
 
 void SpinBox::_line_edit_focus_exit() {
 	// Discontinue because the focus_exit was caused by left-clicking the arrows.
-	if (get_viewport()->gui_get_focus_owner() == get_line_edit()) {
+	Viewport *viewport = get_viewport();
```
Comment (KoBeWi) [inline scene/gui/spin_box.cpp:205]:
```suggestion
	const Viewport *viewport = get_viewport();
```

**Review by KoBeWi — APPROVED** [review]

File: scene/gui/spin_box.cpp (hunk 1 above)
Comment (aaronfranke) [reply scene/gui/spin_box.cpp]: In order do to this I also had to make `Viewport::gui_get_focus_owner()` be const.

**Comment by RedworkDE** [comment]
More changes that are necessary because of the `const`:

```
GH-77757
--------
Validate extension JSON: Error: Field 'classes/Viewport/methods/gui_get_focus_owner': is_const changed value in new API, from false to true.
Validate extension JSON: Error: Hash changed for 'classes/Viewport/methods/gui_get_focus_owner', from 31757941 to A5E188F5. This means that the function has changed and no compatibility function was provided.
```
needs to be added to `misc/extension_api_validation/4.0-stable.expected` (And theoretically a compatibility bind needs to be added as well, but the format for it is not yet fixed, so I will add it in the compat methods PR)

but that should be it for the rats tail of changes.

**Comment by aaronfranke** [comment]
@RedworkDE Thanks, good catch. Do you think we can add a CI check for this? Also, [related](https://www.youtube.com/watch?v=AbSehcT19u0).

**Comment by RedworkDE** [comment]
There is a CI check for that: https://github.com/godotengine/godot/actions/runs/5157156471/jobs/9289181747#step:21:17, it just shows up as an error annotation and not a build failure tho.

**Review by RedworkDE — APPROVED** [review]
The bits of this PR that aren't the crash fix look good to me, and the actual fix also makes sense.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**